docs/guides/Redux-Integration.md

```diff
@@ -85,3 +85,35 @@ To make jest tests work with your react-navigation app, you need to change the j
   ]
 }
 ```
+
+## Extra : Handling back button in android
+
+By using the following snippet, your nav component will be aware of the back button press actions and will correctly interact with your stack. This is really useful on Android.
+
+```javascript
+class ReduxNavigation extends React.Component {
+  componentDidMount() {
+    BackHandler.addEventListener("hardwareBackPress", this.onBackPress);
+  }
+  componentWillUnmount() {
+    BackHandler.removeEventListener("hardwareBackPress", this.onBackPress);
+  }
+  onBackPress = () => {
+    const { dispatch, nav } = this.props;
+    if (nav.index === 0) {
+      return false;
+    }
+    dispatch(NavigationActions.back());
+    return true;
+  };
+
+  render() {
+    const { dispatch, nav } = this.props;
+    const navigation = addNavigationHelpers({
+      dispatch,
+      state: nav
+    });
+
+    return <AppNavigation navigation={navigation} />;
```

Comment:
Can you declare AppNavigation somewhere? This helps those who might not know what exactly you are doing here.